### PR TITLE
Add support for IAwaitOperation to control flow graph

### PR DESCRIPTION
**Pull request by chborl**
@AlekseyTs , @mavasani @jaredpar @333fred for review.

### CODE REVIEW

**Comment by chborl** [comment]
@dotnet-bot retest windows_debug_unit32_prtest please

**Comment by chborl** [comment]
@dotnet-bot retest windows_debug_unit64_prtest please

**Comment by chborl** [comment]
@dotnet-bot retest windows_release_unit32_prtest please

**Comment by chborl** [comment]
@dotnet-bot retest windows_release_unit32_prtest please

**Comment by chborl** [comment]
@dotnet-bot retest windows_release_vs-integration_prtest please

File: src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_IAwaitExpression.cs [hunk 1]
```diff
@@ -223,5 +223,216 @@ static void M(Task<int> t)
 
             VerifyOperationTreeAndDiagnosticsForTest<LocalDeclarationStatementSyntax>(source, expectedOperationTree, expectedDiagnostics, useLatestFrameworkReferences: true);
         }
+
+        [CompilerTrait(CompilerFeature.IOperation, CompilerFeature.Dataflow)]
```
Comment (333fred) [inline src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_IAwaitExpression.cs]: I don't see any tests for when the control flow is the direct target of the await. For example, `var i = await (b ? AsyncMethod1() : AsyncMethod2());`

Comment (AlekseyTs) [reply src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_IAwaitExpression.cs]: > I don't see any tests for when the control flow is the direct target of the await. For example, var i = await (b ? AsyncMethod1() : AsyncMethod2());

I do not think this test would bring any additional value by comparison to AwaitFlow_03

---
In reply to: [188093433](https://github.com/dotnet/roslyn/pull/26811#discussion_r188093433) [](ancestors = 188093433)

**Pull request merged**